common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/MqttTransportHandler.java

```diff
@@ -216,22 +243,28 @@ private void handleGatewayPublishMsg(String topicName, int msgId, MqttPublishMes
     private void processDevicePublish(ChannelHandlerContext ctx, MqttPublishMessage mqttMsg, String topicName, int msgId) {
         try {
             if (deviceSessionCtx.isDeviceTelemetryTopic(topicName)) {
-                TransportProtos.PostTelemetryMsg postTelemetryMsg = adaptor.convertToPostTelemetry(deviceSessionCtx, mqttMsg);
+                TransportProtos.PostTelemetryMsg postTelemetryMsg = deviceSessionCtx.isJsonPayloadType() ?
```

Comment:
var adaptor = deviceSessionCtx.getPayloadAdaptor(). Then you don't need to do a copy-paste all the time.